xla/service/gpu/gpu_compiler_test.cc

```diff
@@ -1099,6 +1099,19 @@ ENTRY main {
   VerifyPassOrder(passes, "layout-assignment", "layout_normalization");
 }
 
+TEST_F(GpuCompilerTest, IndexZeroClampingWorksCorrectly) {
```

Comment:
Personally I would prefer if this is filecheck test in the xla/service/gpu/fusions/test subdirectory. Note that these tests have a way to do a RunAndCompare call through test_correctness.

Reply:
Makes sense. I tried that and fusion_to_mlir from clean XLA crashes on this HLO with:

```
Program received signal SIGSEGV, Segmentation fault.
0x00005555576ef5bc in mlir::detail::OperandStorage::OperandStorage(mlir::Operation*, mlir::OpOperand*, mlir::ValueRange) ()
(gdb) bt
#0  0x00005555576ef5bc in mlir::detail::OperandStorage::OperandStorage(mlir::Operation*, mlir::OpOperand*, mlir::ValueRange) ()
#1  0x00005555576e6d84 in mlir::Operation::create(mlir::Location, mlir::OperationName, mlir::TypeRange, mlir::ValueRange, mlir::DictionaryAttr, mlir::OpaqueProperties, mlir::BlockRange, unsigned int) ()
#2  0x00005555576e6853 in mlir::Operation::create(mlir::Location, mlir::OperationName, mlir::TypeRange, mlir::ValueRange, mlir::NamedAttrList&&, mlir::OpaqueProperties, mlir::BlockRange, mlir::RegionRange) ()
#3  0x00005555576e6729 in mlir::Operation::create(mlir::OperationState const&) ()
#4  0x00005555565250f4 in void mlir::OpBuilder::createOrFold<mlir::arith::AddIOp, mlir::Value&, mlir::Value>(llvm::SmallVectorImpl<mlir::Value>&, mlir::Location, mlir::Value&, mlir::Value&&) ()
#5  0x000055555651a827 in xla::gpu::mlir_converter::(anonymous namespace)::HloToMlir(xla::HloInstruction const*, mlir::func::FuncOp, mlir::ValueRange, std::function<absl::lts_20230802::StatusOr<llvm::SmallVector<mlir::Value, 1u> > (xla::HloInstruction const*, int, mlir::ValueRange)> const&, std::function<mlir::func::FuncOp (xla::HloInstruction const*)> const&, mlir::ImplicitLocOpBuilder&) ()
#6  0x0000555556516c31 in xla::gpu::mlir_converter::(anonymous namespace)::SubgraphConverter::EmitInstruction(xla::HloInstruction const*, mlir::ValueRange) ()
#7  0x0000555556513b03 in xla::gpu::mlir_converter::SubgraphToMlirFunction(xla::gpu::mlir_converter::PartitionedComputation const&, xla::gpu::mlir_converter::PartitionedComputation::Subgraph const&, mlir::func::FuncOp&, std::function<mlir::func::FuncOp (xla::HloInstruction const*)> const&) ()
#8  0x00005555564be5eb in xla::gpu::MlirFusionEmitterBase::EmitMlir(mlir::ModuleOp, mlir::func::FuncOp, xla::HloFusionInstruction const&) const ()
#9  0x00005555564bc2d0 in xla::gpu::MlirFusionEmitterBase::CreateMLIRModule(mlir::MLIRContext&, xla::HloFusionInstruction const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, xla::BufferAssignment const*, mlir::interpreter::MlirCompilationTrace*) const ()
#10 0x00005555560a22e7 in xla::gpu::Run(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#11 0x00005555560a2622 in main ()

```